src/hotspot/os/linux/cgroupSubsystem_linux.cpp

```diff
@@ -63,10 +64,11 @@ CgroupSubsystem* CgroupSubsystemFactory::create() {
     // Construct the subsystem, free resources and return
     // Note: any index in cg_infos will do as the path is the same for
     //       all controllers.
-    CgroupController* unified = new CgroupV2Controller(cg_infos[MEMORY_IDX]._mount_path, cg_infos[MEMORY_IDX]._cgroup_path);
+    CgroupV2MemoryController* memory = new CgroupV2MemoryController(cg_infos[MEMORY_IDX]._mount_path, cg_infos[MEMORY_IDX]._cgroup_path);
```

Comment:
```suggestion
    CgroupV2MemoryController* memory = new CgroupV2MemoryController(cg_infos.memory._mount_path, cg_infos.memory._cgroup_path);
```
First could there be:
class CgroupInfos {
  CgroupInfo[5];
  CgroupInfo &cpuset = CgroupInfo[0];
  CgroupInfo &cpu = CgroupInfo[1];
  etc.
};
and remove:
#define CG_INFO_LENGTH 5
#define CPUSET_IDX     0
#define CPU_IDX        1
...
Second the constructor should have `const CgroupInfo &` as a parameter.

Reply:
Feel free to propose this cleanup as a separate patch. I'd like to keep this one on-topic to do controller delegation if that's ok.